src/Jeffijoe.MessageFormat/CustomValueFormatters.cs

```diff
@@ -0,0 +1,209 @@
+using System.Diagnostics.CodeAnalysis;
+using System.Globalization;
+
+namespace Jeffijoe.MessageFormat;
+
+/// <summary>
+///     Attempts to format a date.
+/// </summary>
+/// <param name="culture">
+///     The culture.
+/// </param>
+/// <param name="value">
+///     The value to format.
+/// </param>
+/// <param name="style">
+///     The requested style, if any.
+/// </param>
+/// <param name="formatted">
+///     Output for setting the formatted result.
+/// </param>
+/// <returns>
+///     <value>true</value> if able to format the value; <value>false</value> otherwise.
+/// </returns>
+public delegate bool TryFormatDate(
+    CultureInfo culture,
+    object? value,
+    string? style,
+    out string? formatted);
```

Comment:
That only works on .NET 5 and above, and since we are multi-targeting, it would require a lot of `#ifdef`s. I am thinking once .NET 8 drops, I will bump the minimum supported TFMs so we don't need conditionals anymore.

Would you happen to know what the most common minimum version to support for libraries is these days?

Reply:
[This document](https://learn.microsoft.com/en-us/dotnet/api/system.diagnostics.codeanalysis.notnullwhenattribute?view=net-7.0) suggests that it's available from .NET Core 3.0, but I guess it's still an issue since you'd want to also support .NET Framework.

I [don't believe](https://github.com/VerifyTests/Verify/issues/784#issuecomment-1460683975) in supporting EOL frameworks, but .NET Framework 4.6.2+ is [still supported](https://learn.microsoft.com/en-us/lifecycle/products/microsoft-net-framework#releases). With that said, both [Verify](https://github.com/VerifyTests/Verify/issues/784) and [xUnit v3](https://xunit.net/docs/v3-alpha#v2-changes) only support 4.7.2+.

A [similar discussion](https://github.com/serilog/serilog/issues/1820#issuecomment-1533865726) for Serilog, though I [don't agree](https://github.com/serilog/serilog/issues/1820#issuecomment-1534192646) with including the EOL net5.0.

Reply:
I think `netstandard2.1` has everything we need, no? Right now we are targeting standard 2.0.

Reply:
My understanding of .NET Standard is not perfect, but I think that if you drop `netstandard2.0`, then the library would have no support for .NET Framework, unless you also specified the supported versions, like Verify does ...

```xml
<TargetFrameworks>net472;net48;net6.0;net7.0;net8.0</TargetFrameworks>
```

([ref](https://github.com/VerifyTests/Verify/pull/981/files#r1311209866))

Reply:
Oh neat! Will those give us all the features we need? `[NotNullWhen]` and `Span<T>`?

Reply:
No. You could use [Polyfill](https://github.com/SimonCropp/Polyfill) to enable `[NotNullWhen]` (probably not worth an extra dependency), but `Span<T>` likely won't be possible as long as you support .NET Framework.

Reply:
I am almost inclined to say let's just stop supporting .NET Framework.